LEDGAME.ino

```diff
@@ -75,140 +79,216 @@ int randomGenTime()
   return randomtime;
 }
 
- //turn on random led
-int randomLedOn(int num)
+ //turn on led
+int ledOn(int num)
 {
-   digitalWrite(ledInit[num], HIGH);   
+  digitalWrite(ledInit[num], HIGH);   
 }
 
-
 //turn off all leds
 void ledOff(int num)
+{
+  digitalWrite(ledInit[num], LOW);
+}  
+
+//butt buffer 
+int checkButtPress()
+{  
+  for (int i = 1; i < BUTTONS_QUANTITY; i++)
   {
-    digitalWrite(ledInit[num], LOW);
+    if (digitalRead(buttInit[i]) == HIGH)
+    {
+      buttState[i] = 1;
+      buttPressStatus = true;
+    }
   }  
+}
 
-//буфер с нажатыми кнопками
-int WhichButtPressed()
-{
-  
-  for (int c = 1; c < 4; c++)
+int checkResult(int num)
+{ 
+ if (buttState[num] == 1 && buttPressStatus)
+  {
+    score += 1;
+    lcd.clear();
+    lcd.print("    CORRECT!");
+  }
+ else if (buttState[num] == 0 && buttPressStatus)
+  {
+    lcd.clear();
+    lcd.print("   INCORRECT(");
+  }
+ else if (!ledTimer.isTimerRunning() && !buttPressStatus)
   {
-   if (digitalRead(buttInit[c]) == HIGH)
+    lcd.clear();
+    lcd.print("NO BUTTON PRESS(");
+  }
+}
+
+//clearing butt bufer
+void clearButt()
+{
+  buttState[0] = 1; 
+  buttPressStatus = false;   
+  for (int i = 1; i < BUTTONS_QUANTITY; i++)
    {
-     butt[c] = c;
-     i = 1;
+     buttState[i] = 0;
    }
-  } 
-  
 }
 
-
-int Logic(int num)
+ //waiting for start button
+void startButtonWait()
 {
- if (butt[num] == num && i == 1)
- {
-  score += 1;
-  Serial.print("                                                                  CORRECT! \n");
-  //correct
- }
-  else if (butt[num] != num && i == 1)
+  while (buttState[0] == 0) 
   {
-    Serial.print("                                                                 INCORRECT( \n");
-    //incorrect
+    if(digitalRead(buttInit[0]) == HIGH)
+    { 
+      buttState[0] = 1; 
+    }
   }
-   else if (ledTimer.isTimerRunning() == false && i == 0)
-   {
-    Serial.print("                                                          NO BUTTON PRESS DETECTED \n");
-   }
 }
 
-//clearing butt bufer
-    void clearButt()
+void userSetGameTime()

```

Comment:
this function is also OK, but it uses only 3 buttons and if we will use 4 or more buttons - it will not work correctly! 
I mean it's better to check only needed buttons, not all buttons